src/main/java/gregtech/common/tileentities/machines/multi/compressor/GT_MetaTileEntity_HIPCompressor.java

```diff
@@ -0,0 +1,524 @@
+
+                if (cooling) {
+                    setSpeedBonus(2.5F);
+                    setEuModifier(1.1F);
+                }
+
+                if (recipe.mSpecialValue > 0) doingHIP = true;
+                return super.validateRecipe(recipe);
+            }
+        }.setMaxParallelSupplier(this::getMaxParallelRecipes);
+    }
+
+    @Override
+    public boolean onRunningTick(ItemStack aStack) {
+        if (cooling && doingHIP) {
+            stopMachine(SimpleShutDownReason.ofCritical("overheated"));
+            doingHIP = false;
+        }
+        return super.onRunningTick(aStack);
+    }
+
+    @Override
+    public void onPostTick(IGregTechTileEntity aBaseMetaTileEntity, long aTick) {
+        super.onPostTick(aBaseMetaTileEntity, aTick);
+
+        if (aTick % 20 == 0) {
+
+            // Default to cooling by 2%
+            float heatMod = -2;
+
+            // If the machine is running, heat by 5% x 0.90 ^ (Coil Tier)
+            // Cupronickel is 0, so base will be 5% increase
+            if (this.maxProgresstime() != 0) {
+                heatMod = (float) (5 * Math.pow(0.9, coilTier));
+            }
+
+            heat = MathUtils.clamp(heat + heatMod, 0, 100);
+
+            if ((cooling && heat <= 0) || (!cooling && heat >= 100)) {
+                cooling = !cooling;
+            }
+        }
+
+        // Update all the sensors
+        for (GT_MetaTileEntity_HeatSensor hatch : sensorHatches) {
+            hatch.updateRedstoneOutput(heat);
+        }
+
+    }
+
+    public int getMaxParallelRecipes() {
+        return cooling ? 1 : (8 * GT_Utility.getTier(this.getMaxInputVoltage()));
+    }
+
+    @Override
+    public RecipeMap<?> getRecipeMap() {
+        return RecipeMaps.compressorRecipes;
+    }
+
+    @Override
+    public int getMaxEfficiency(ItemStack aStack) {
+        return 10000;
+    }
+
+    @Override
+    public int getDamageToComponent(ItemStack aStack) {
+        return 0;
+    }
+
+    @Override
+    public boolean explodesOnComponentBreak(ItemStack aStack) {
+        return false;
+    }
+
+    @Override
+    public boolean supportsVoidProtection() {
+        return true;
+    }
+
+    @Override
+    public boolean supportsBatchMode() {
+        return true;
+    }
+
+    @Override
+    public boolean supportsInputSeparation() {
+        return true;
+    }
+
+    @Override
+    public boolean supportsSingleRecipeLocking() {
+        return true;
+    }
+
+    public HeatingCoilLevel getCoilLevel() {
+        return heatLevel;
+    }
+
+    public void setCoilLevel(HeatingCoilLevel aCoilLevel) {
+        heatLevel = aCoilLevel;
+        coilTier = aCoilLevel.getTier();
+    }
+
+    public boolean addSensorHatchToMachineList(IGregTechTileEntity aTileEntity, int aBaseCasingIndex) {
+        if (aTileEntity == null) return false;
+        IMetaTileEntity aMetaTileEntity = aTileEntity.getMetaTileEntity();
+        if (aMetaTileEntity instanceof GT_MetaTileEntity_HeatSensor) {
```

Comment:
Use modern `instanceof` cast thing.

```suggestion
        if (aMetaTileEntity instanceof GT_MetaTileEntity_HeatSensor sensor) {
```